controller/service.rst

```diff
@@ -22,8 +22,26 @@ in method parameters:
         # controllers are imported separately to make sure services can be injected
         # as action arguments even if you don't extend any base controller class
         App\Controller\:
-           resource: '../src/Controller/'
-           tags: ['controller.service_arguments']
+            resource: '../src/Controller/'
+            tags: ['controller.service_arguments']
+
+.. note::
+
+    If you don't use either :doc:`autowiring </service_container/autowiring>`
+    or :ref:`autoconfiguration <services-autoconfigure>`, you'll need to apply
+    other tags and make some method calls to register your controllers as services:
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+
+        # this extended configuration is only required when not using autowiring/autoconfiguration,
+        # which is uncommon and not recommended
+        App\Controller\:
+            resource: '../src/Controller/'
+            tags: ['controller.service_arguments', 'container.service_subscriber']
+            calls:
+                - setContainer: ['@service_container']
```

Comment:
are you sure service_container is correct here as one needs only the service locator instance of the service subscriber and not the full container or is the service ID replaced here in case the container.service_subscriber is used? 

Also I think if the service subscriber tag is active it should set the correct container service even without autowiring but I am not sure.

Reply:
With the config shown, it worked for me when disabling autowire/autoconfigure. Other configs didn't work for me.

If you can, please tell me the exact config to try and I'll report the results here. Thanks!

Reply:
hi @javiereguiluz it is possible that it is correct but e.g. in other bundles like the user bundle https://github.com/FriendsOfSymfony/FOSUserBundle/blob/master/Resources/config/registration.xml#L26 this was used for the call.

Reply:
@javiereguiluz I have now tested it and with Symfoyn 5.4 at least, if I use 

```yaml
calls:
    - setContainer: [ '@Psr\Container\ContainerInterface' ]
```
then I get `Symfony\Component\DependencyInjection\Argument\ServiceLocator` as class of the container.

If I use 

```yaml
calls:
    - setContainer: ['@service_container']
```

I get `ContainerIjSHgNp\App_KernelDevDebugContainer` as class of the container.


The first one seems to be correct as the second one is the full service container which might actually not have all relevant services.

Reply:
and if I use `$this->container->get('parameter_bag')` with the first it works the second results in service has been inlined and removed.

Reply:
btw. I think the config should only be done for a specific controller class as example, as controllers that do not extends AbstractController might fail otherwise (they might not use setcontainer).

Reply:
I also tried with a minimalistic example:

```php
class TestController extends AbstractController implements ServiceSubscriberInterface
{
    #[Route('/')]
    public function index(): Response
    {
        return new Response();
    }
}
```

Config:

```yaml
services:
    # ...

    App\Controller\:
        resource: '../src/Controller'
        public: true
        autowire: false
        autoconfigure: false
        tags: ['controller.service_arguments', 'container.service_subscriber']
        calls:
            - setContainer: ['@service_container']
```

And I also have the `The "parameter_bag" service or alias has been removed or inlined when the container was compiled` error :thinking: 

I wonder if it isn't related to [this known bug](https://github.com/symfony/symfony/issues/49082) actually.

Reply:
I think that at this point, we should ask @nicolas-grekas for advice about what to do here. Thanks!

Reply:
There is never a reason to inject the service container anywhere, so this is wrong. It will fail when any private services are accessed. It happens that AbstractController consumes services that are declared public (the router?) But this works by chance. Instead one should declare a scoped locator. There's a way to define one using tags, I don't remember how exactly and I'm AFK. Anyone does?

Reply:
The locator should be defined by the container.servicer_subscriber tag in this case as the Abstractconroller provides the corresponding subscribedservices method.  otherwise it is even more configuration to get all correct arguments into the abstractcontroller (which might even change in the future).
not sure what the ID of that one is to inject it. I think however, ContainerInterface should work because that is what autowiring binds the service for - otherwise it would not work correctly with autowiring but not sure if this is the preferred service ID.

might also be something to add to the service subscriber/locator documentation because I don't think it is mentioned there either.